sharding-core/sharding-core-route/src/main/java/org/apache/shardingsphere/core/route/router/masterslave/MasterSlaveRouter.java

```diff
@@ -70,6 +72,14 @@
     }
     
     private boolean isMasterRoute(final SQLStatement sqlStatement) {
-        return !(sqlStatement instanceof SelectStatement) || MasterVisitedManager.isMasterVisited() || HintManager.isMasterRouteOnly();
+        return isLockSegment(sqlStatement) || !(sqlStatement instanceof SelectStatement) || MasterVisitedManager.isMasterVisited() || HintManager.isMasterRouteOnly();
+    }
+
+    private boolean isLockSegment(final SQLStatement sqlStatement) {
```

Comment:
maybe we can simplify the code as:

`return sqlStatement instanceof SelectStatement && ((SelectStatement) sqlStatement).getLock().isPresent()`